### PR TITLE
vtk-m +cuda: conflicts with cuda_arch=none

### LINKED ISSUE #27915: vtk-m +cuda build fails when cuda_arch=none: VTKmDeviceAdapters.cmake:224: set VTKm_CUDA_Architecture manually
### Steps to reproduce the issue

`vtk-m@1.6.0 +cuda` build fails if `cuda_arch=none`

Using:
* `spack@develop` (f81d84dfc6b1c3a821dffab785e62ecd79f130ab from `Tue Dec 7 12:30:14 2021 +0100`)
* Ubuntu 20.04, GCC 9.3.0, X86_64

Concrete spec: [vtk-m.spec.json.txt](https://github.com/spack/spack/files/7694066/vtk-m.spec.json.txt)

Concretization:
```
vtk-m@1.6.0%gcc@9.3.0~64bitids+ascent_types+cuda+doubleprecision~hip~ipo~logging+mpi+openmp+rendering~shared~tbb~virtuals amdgpu_target=none build_type=Release cuda_arch=none arch=linux-ubuntu20.04-x86_64
    ^cmake@3.21.4%gcc@9.3.0~doc+ncurses+openssl+ownlibs~qt build_type=Release arch=linux-ubuntu20.04-x86_64
        ^ncurses@6.2%gcc@9.3.0~symlinks+termlib abi=none arch=linux-ubuntu20.04-x86_64
            ^pkgconf@1.8.0%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
        ^openssl@1.1.1l%gcc@9.3.0~docs certs=system arch=linux-ubuntu20.04-x86_64
            ^perl@5.34.0%gcc@9.3.0+cpanm+shared+threads arch=linux-ubuntu20.04-x86_64
                ^berkeley-db@18.1.40%gcc@9.3.0+cxx~docs+stl patches=b231fcc4d5cff05e5c3a4814f6a5af0e9a966428dc2176540d2c05aff41de522 arch=linux-ubuntu20.04-x86_64
                ^bzip2@1.0.8%gcc@9.3.0~debug~pic+shared arch=linux-ubuntu20.04-x86_64
                    ^diffutils@3.8%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                        ^libiconv@1.16%gcc@9.3.0 libs=shared,static arch=linux-ubuntu20.04-x86_64
                ^gdbm@1.19%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                    ^readline@8.1%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                ^zlib@1.2.11%gcc@9.3.0+optimize+pic+shared arch=linux-ubuntu20.04-x86_64
    ^cuda@11.5.1%gcc@9.3.0~dev arch=linux-ubuntu20.04-x86_64
        ^libxml2@2.9.12%gcc@9.3.0~python arch=linux-ubuntu20.04-x86_64
            ^xz@5.2.5%gcc@9.3.0+pic libs=shared,static arch=linux-ubuntu20.04-x86_64
    ^mpich@3.4.2%gcc@9.3.0~argobots+fortran+hwloc+hydra+libxml2+pci+romio~slurm~two_level_namespace~verbs~wrapperrpath device=ch4 netmod=ofi pmi=pmi arch=linux-ubuntu20.04-x86_64
        ^findutils@4.8.0%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
        ^hwloc@2.6.0%gcc@9.3.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~opencl+pci~rocm+shared arch=linux-ubuntu20.04-x86_64
            ^libpciaccess@0.16%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                ^libtool@2.4.6%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                    ^m4@1.4.19%gcc@9.3.0+sigsegv patches=9dc5fbd0d5cb1037ab1e6d0ecc74a30df218d0a94bdd5a02759a97f62daca573,bfdffa7c2eb01021d5849b36972c069693654ad826c1a20b53534009a4ec7a89 arch=linux-ubuntu20.04-x86_64
                        ^libsigsegv@2.13%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                ^util-macros@1.19.3%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
        ^libfabric@1.14.0%gcc@9.3.0~debug~kdreg fabrics=rxm,sockets,tcp,udp arch=linux-ubuntu20.04-x86_64
```

Error:
```
$> spack install -f ./vtk-m.spec.json
...
==> Installing vtk-m-1.6.0-uczqqylkiff7uob3g4e4nic4kcwejx7i
==> No binary for vtk-m-1.6.0-uczqqylkiff7uob3g4e4nic4kcwejx7i found: installing from source
==> Using cached archive: /spack/var/spack/cache/_source-cache/archive/14/14e62d306dd33f82eb9ddb1d5cee987b7a0b91bf08a7a02ca3bce3968c95fd76.tar.gz
==> No patches needed for vtk-m
==> vtk-m: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    'cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/vtk-m-1.6.0-uczqqylkiff7uob3g4e4nic4kcwejx7i' '-DCMAKE_BUILD_TYPE:STRING=Release' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=OFF' '-DCMAKE_INSTALL_RPATH:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/vtk-m-1.6.0-uczqqylkiff7uob3g4e4nic4kcwejx7i/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/vtk-m-1.6.0-uczqqylkiff7uob3g4e4nic4kcwejx7i/lib64;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libxml2-2.9.12-xlbi5rk675mptoyjnuff4q3e2lajs2wr/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libiconv-1.16-zth4ohvj7y3p532iqjw7fptg3kkn3ixv/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/xz-5.2.5-mq4ikt5q4zttamuxqkqe66jlftjbvkf4/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/zlib-1.2.11-ufqpqnpj5gjkfx5sd2o3twwveuowcz2b/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mpich-3.4.2-da2intlyhnltoebtrcvfl7qeffmwwzp6/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hwloc-2.6.0-bivhc36xulak7wjqfzshvzhioqs3ki77/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libpciaccess-0.16-4jbn7mznw664dc4mlseouzovsfx6sor5/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/ncurses-6.2-nxjjphalwbf2ywlxpn74nsfcsrq2bbtd/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libfabric-1.14.0-ohufqpc3kq5sd32tcn2pp3t636knmxxd/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/cuda-11.5.1-7zovxzh23daik22jdh4kb5l4gobp5ojt/lib64' '-DCMAKE_PREFIX_PATH:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mpich-3.4.2-da2intlyhnltoebtrcvfl7qeffmwwzp6;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libfabric-1.14.0-ohufqpc3kq5sd32tcn2pp3t636knmxxd;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hwloc-2.6.0-bivhc36xulak7wjqfzshvzhioqs3ki77;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libpciaccess-0.16-4jbn7mznw664dc4mlseouzovsfx6sor5;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/cuda-11.5.1-7zovxzh23daik22jdh4kb5l4gobp5ojt;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libxml2-2.9.12-xlbi5rk675mptoyjnuff4q3e2lajs2wr;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/xz-5.2.5-mq4ikt5q4zttamuxqkqe66jlftjbvkf4;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/cmake-3.21.4-55vqjfj2bk55ga4fj3erj5tpzugh3t3b;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/zlib-1.2.11-ufqpqnpj5gjkfx5sd2o3twwveuowcz2b;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libiconv-1.16-zth4ohvj7y3p532iqjw7fptg3kkn3ixv;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/ncurses-6.2-nxjjphalwbf2ywlxpn74nsfcsrq2bbtd' '-DVTKm_ENABLE_TESTING:BOOL=OFF' '-DBUILD_SHARED_LIBS=OFF' '-DVTKm_USE_DOUBLE_PRECISION:BOOL=ON' '-DVTKm_ENABLE_LOGGING:BOOL=OFF' '-DVTKm_ENABLE_MPI:BOOL=ON' '-DVTKm_ENABLE_RENDERING:BOOL=ON' '-DVTKm_USE_DEFAULT_TYPES_FOR_ASCENT:BOOL=ON' '-DVTKm_NO_DEPRECATED_VIRTUAL:BOOL=ON' '-DVTKm_USE_64BIT_IDS:BOOL=OFF' '-DVTKm_ENABLE_CUDA:BOOL=ON' '-DCMAKE_CUDA_HOST_COMPILER=/usr/bin/g++' '-DVTKm_ENABLE_HIP:BOOL=OFF' '-DVTKm_ENABLE_OPENMP:BOOL=ON' '-DVTKm_ENABLE_TBB:BOOL=OFF' '/tmp/root/spack-stage/spack-stage-vtk-m-1.6.0-uczqqylkiff7uob3g4e4nic4kcwejx7i/spack-src'

1 error found in build log:
     17    -- The CUDA compiler identification is NVIDIA 11.5.119
     18    -- Detecting CUDA compiler ABI info
     19    -- Detecting CUDA compiler ABI info - done
     20    -- Check for working CUDA compiler: /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/cuda-11.5.1-7zovxzh23daik22jdh4kb5l4gobp5ojt/bin/nvcc
           - skipped
     21    -- Detecting CUDA compile features
     22    -- Detecting CUDA compile features - done
  >> 23    CMake Error at CMake/VTKmDeviceAdapters.cmake:224 (message):
     24      Error detecting architecture flags for CUDA.  Please set
     25      VTKm_CUDA_Architecture manually.
     26    Call Stack (most recent call first):
     27      CMake/VTKmWrappers.cmake:14 (include)
     28      CMakeLists.txt:179 (include)
```

### Information on your system

* **Spack:** 0.17.0-445-f81d84dfc6
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-cascadelake
* **Concretizer:** clingo

### Additional information

[spack-build-01-cmake-out.txt](https://github.com/spack/spack/files/7694086/spack-build-01-cmake-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/7694087/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/7694088/spack-build-out.txt)

@kmorel
@robertmaynard 
@vicentebolea

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

### PR DESCRIPTION
**Pull request by eugeneswalker**
`vtk-m +cuda` conflicts with `cuda_arch=none`

Fixes https://github.com/spack/spack/issues/27915

### CODE REVIEW

**Review by kmorel — APPROVED** [review]
+1

**Comment by eugeneswalker** [comment]
Should we merge now or wait for review from other maintainers?

**Comment by vicentebolea** [comment]
@eugeneswalker
> Should we merge now or wait for review from other maintainers?

Yes, thanks for the contribution :)

**Review by wspear — APPROVED** [review]
Looks good!

**Pull request merged**
